omni-relayer/src/utils/near.rs

```diff
@@ -0,0 +1,117 @@
+    IndexerExecutionOutcomeWithReceipt, StreamerMessage,
+};
+use omni_types::near_events::Nep141LockerEvent;
+
+use crate::{config, utils};
+
+pub async fn get_final_block(jsonrpc_client: &JsonRpcClient) -> Result<u64> {
+    info!("Getting final block");
+
+    let block_response = RpcBlockRequest {
+        block_reference: near_primitives::types::BlockReference::Finality(
+            near_primitives::types::Finality::Final,
+        ),
+    };
+    jsonrpc_client
+        .call(block_response)
+        .await
+        .map(|block| block.header.height)
+        .map_err(Into::into)
+}
+
+pub async fn handle_streamer_message(
+    config: &config::Config,
+    redis_connection: &mut redis::aio::MultiplexedConnection,
+    jsonrpc_client: &JsonRpcClient,
+    streamer_message: &StreamerMessage,
+) {
+    let nep_locker_event_outcomes = find_nep_locker_event_outcomes(config, streamer_message);
+
+    let nep_locker_event_logs = nep_locker_event_outcomes
+        .iter()
+        .flat_map(|outcome| outcome.execution_outcome.outcome.logs.clone())
+        .filter_map(|log| serde_json::from_str::<Nep141LockerEvent>(&log).ok())
+        .collect::<Vec<_>>();
+
+    for log in nep_locker_event_logs {
+        info!("Processing Nep141LockerEvent: {:?}", log);
+
+        match log {
+            Nep141LockerEvent::InitTransferEvent {
+                ref transfer_message,
+            } => {
+                // TODO: If fee is insufficient, it should be handled later. For example,
+                // add to redis and try again in 1 hour
+                match utils::price::is_fee_sufficient(
+                    jsonrpc_client,
+                    &transfer_message.sender,
+                    &transfer_message.recipient,
+                    &transfer_message.token,
+                    transfer_message.fee.into(),
+                )
+                .await
+                {
+                    Ok(res) => {
+                        if !res {
+                            warn!("Fee is insufficient");
+                        }
+                    }
+                    Err(err) => {
+                        warn!("Failed to check fee: {}", err);
+                    }
+                }
+
+                utils::redis::add_event(
+                    redis_connection,
+                    &config.redis.near_init_transfer_events,
+                    transfer_message.origin_nonce.0.to_string(),
+                    log,
+                )
+                .await;
+            }
+            Nep141LockerEvent::SignTransferEvent {
+                ref message_payload,
+                ..
+            } => {
+                utils::redis::add_event(
+                    redis_connection,
+                    &config.redis.near_sign_transfer_events,
+                    message_payload.nonce.0.to_string(),
+                    log,
+                )
+                .await;
+            }
+            Nep141LockerEvent::FinTransferEvent { .. }
```

Comment:
This event should be handled the same way as InitTransferEvent in some cases.

Reply:
InitTransfer should trigger `sign_transfer` method, but why do we need to sign anything after the transfer being finalized? In future we can add `claim_fee` that would claim fee on the sender's chain, but it has nothing to do with signing something

I don't see any extra steps from relayer after getting a FinTransferEvent in this graph as well: https://docs.nearone.org/doc/omni-bridge-IBwRPSc6SG#h-eth-→-near-transfer. We just need to catch `Withdraw` event and call `fin_transfer` right after